metrics-operator/controllers/common/providers/elk/elk.go

```diff
@@ -0,0 +1,151 @@
+package elasticsearch
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	elastic "github.com/elastic/go-elasticsearch/v8"
+	"github.com/go-logr/logr"
+	metricsapi "github.com/keptn/lifecycle-toolkit/metrics-operator/api/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	warningLogStringElastic = "%s API returned warnings: %s"
+	defaultTimeRange        = 30 * time.Minute
+)
+
+type KeptnElasticProvider struct {
+	Log       logr.Logger
+	K8sClient client.Client
+	Elastic   *elastic.Client
+}
+
+type ElasticsearchResponse struct {
+	Hits struct {
+		Total struct {
+			Value int `json:"value"`
+		} `json:"total"`
+	} `json:"hits"`
+}
+
+func NewElasticProvider(log logr.Logger, k8sClient client.Client, elasticURL string) (*KeptnElasticProvider, error) {
+	cfg := elastic.Config{
+		Addresses: []string{
+			elasticURL,
+		},
+	}
+	es, err := elastic.NewClient(cfg)
```

Comment:
does elastic have a mock library? if yes, let's try to use it for testing. if not, maybe we can create an interface and try to mock the elastic to improve the testing, WDYT?